examples/l2switch/l2switch.c

```diff
@@ -0,0 +1,353 @@
+ *
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *   "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *   A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *   OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *   DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *   THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * l2switch.c - send all packets from one port out the adjacent port.
+ ********************************************************************/
+
+
+#include <errno.h>
+#include <getopt.h>
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/queue.h>
+#include <unistd.h>
+
+#include <rte_common.h>
+#include <rte_ip.h>
+#include <rte_mbuf.h>
+#include <rte_malloc.h>
+
+#include "onvm_nflib.h"
+#include "onvm_pkt_helper.h"
+
+#define NF_TAG "l2switch"
+
+/* Shared data structure containing host port info. */
+extern struct port_info *ports;
+
+/* Per-port statistics struct. */
+struct l2fwd_port_statistics {
+	uint64_t tx;
+	uint64_t rx;
+	uint64_t dropped;
+} __rte_cache_aligned;
+
+/*Struct that holds all NF state information */
+struct state_info {
+       /* Number of package between each print. */
+       uint32_t print_delay;
+       /* List of enabled ports. */
+       uint32_t l2fwd_dst_ports[RTE_MAX_ETHPORTS];
+       /* Ethernet addresses of ports. */
+       struct ether_addr l2fwd_ports_eth_addr[RTE_MAX_ETHPORTS];
+       struct l2fwd_port_statistics port_statistics[RTE_MAX_ETHPORTS];
+       /* MAC updating enabled by default */
+       int mac_updating;
+       /* Print mac address disabled by default */
+       int print_mac;
+};
+
+/*
+ * Print a usage message
+ */
+static void
+usage(const char *progname) {
+        printf("Usage:\n");
+        printf("%s [EAL args] -- [NF_LIB args] -- -p <print_delay>\n", progname);
+        printf("%s -F <CONFIG_FILE.json> [EAL args] -- [NF_LIB args] -- [NF args]\n\n", progname);
+        printf("Flags:\n");
+        printf(" - `-p <print_delay>`: number of packets between each print, e.g. `-p 1` prints every packets.\n");
+        printf(" - `-n : Disables mac updating. \n");
+        printf(" - `-m : Enables printing updated mac address. \n");
+}
+
+/*
+ * Parse the application arguments.
+ */
+static int
+parse_app_args(int argc, char *argv[], const char *progname, struct state_info *stats) {
+        int c;
+
+        while ((c = getopt(argc, argv, "p:n:m")) != -1) {
```

Comment:
I think there shouldn't be a ':' after n here since we don't expect any value to be passed along with `-n` option

Reply:
Thanks!